docs/conf.py

```diff
@@ -42,6 +42,27 @@
 # This pattern also affects html_static_path and html_extra_path.
 exclude_patterns = ["_build", "Thumbs.db", ".DS_Store"]
 
+nitpick_ignore = [
+    ("py:class", "optional"),
+    ("py:class", "abc.ABC"),
+]
+
+nitpick_ignore_regex = [
+    ("py:class", r"algosdk\.abi\..*"),
+    ("py:class", r"contextlib\..*"),
+    ("py:class", r"enum\..*"),
+    ("py.obj", r".*\.T$"),
+    ("py.obj", r".*\.T[0-9]+$"),
+    ("py.class", r".*\.T_co$"),
+    ("py.obj", r".*\.T_co$"),
+    ("py.class", r".*\.T$"),
+    ("py.class", r"^T$"),
+    ("py.obj", r".*\.N$"),
+    ("py.class", r".*\.N$"),
+    ("py.class", r"^N$"),
+    ("py.class", r"^type\[.*"),
+    ("py.obj", r"typing\..*"),
```

Comment:
@ahangsu I opened an optional child PR that removes 2 ignores by upgrading Sphinx:  https://github.com/algorand/pyteal/pull/486.

Can you have a look and see if you feel it's worth pulling in?

I do not feel strongly and will close it if you prefer as is.

Reply:
just approved

Reply:
@ahangsu Thanks for taking in the change - merged.